File: python3/2799.py
class Solution:
    def countCompleteSubarrays(self, n: List[int]) -> int:
        d = 0
        l = len(n)
        c = 0
        t = 0
        p = 0
        f = []
        o = [0] * 2001
        for i in n:
            if not i in f:
                d += 1
                f.append(i)
        for i in range(l):
            if o[n[i]] == 0:
                c += 1
            o[n[i]] += 1
            u = p
            while c >= d:
                o[n[p]] -= 1
                if o[n[p]] == 0:
                    c -= 1
                p += 1
                if c < d:
                    t += (p - u) * (l - i)
                    break
        return t

File: c/2799.c
int countCompleteSubarrays(int* n, int l) {
    int d = 0, c = 0, t = 0, p = 0;
    bool f[2001] = {false};
    int o[2001] = {0};
    for (int i = 0; i < l; i++)
        if (!f[n[i]]) {
            d++;
            f[n[i]] = true;
        }
    for (int i = 0; i < l; i++) {
        if (!o[n[i]])
            c++;
        o[n[i]]++;
        int u = p;
        while (c >= d) {
            o[n[p]]--;
            if (!o[n[p]])
                c--;
            p++;
            if (c < d) {
                t += (p - u) * (l - i);
                break;
            }
        }
    }
    return t;
}
